src/common/telemetry/lib/Telemetry.cpp

```diff
@@ -0,0 +1,269 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "Telemetry.hpp"
+#include "ParameterSets.hpp"
+
+#include <algorithm>
+#include <chrono>
+#include <cstdio>
+#include <cstdlib>
+#include <cstring>
+#include <fstream>
+#include <iostream>
+#include <nlohmann/json.hpp>
+#include <ScopeGuard.h>
+#include <sstream>
+#include <thread>
+
+using namespace MAT;
+
+namespace Telemetry
+{
+
+TelemetryManager::TelemetryManager(bool enableDebug, std::chrono::seconds teardownTime, OsConfigLogHandle logHandle)
+    : m_log(logHandle)
+    , m_logManager(nullptr)
+    , m_logger(nullptr)
+{
+#if defined(DEBUG) || defined(_DEBUG) || !defined(NDEBUG)
+    SetLoggingLevel(LoggingLevelDebug);
```

Comment:
>SetLoggingLevel(LoggingLevelDebug); [](http://example.com/codeflow?start=4&length=35)

Set enableDebug here instead -- why call SetLoggingLervel twice?

Also, why call this at all?

We have a policy parameter to configure the logging level to what the customer wants. This code shall not attempt to overwrite that.